api/agent/agent_test.go

```diff
@@ -123,3 +152,85 @@ func TestHandleDesiredState(t *testing.T) {
 	assert.NoError(t, updAgent.HandleDesiredState(testActivityID, 0, dummyDesiredState))
 	<-ch
 }
+
+func TestHandleDesiredStateCommand(t *testing.T) {
+	mockCtr := gomock.NewController(t)
+	defer mockCtr.Finish()
+
+	mockClient := mocks.NewMockUpdateAgentClient(mockCtr)
+	mockUpdateManager := mocks.NewMockUpdateManager(mockCtr)
+
+	updAgent := &updateAgent{
+		client:  mockClient,
+		manager: mockUpdateManager,
+		ctx:     context.Background(),
+	}
```

Comment:
please also fix TestHandleDesiredState the same way while you are at it
```suggestion
	mockUpdateManager := mocks.NewMockUpdateManager(mockCtr)
	updAgent := &updateAgent{
		client:  mocks.NewMockUpdateAgentClient(mockCtr),
		manager: mockUpdateManager,
		ctx:     context.Background(),
	}
```